shielder/contract/mocked_zk.rs

```diff
@@ -0,0 +1,200 @@
+// mocked zk relation
+
+
+#![cfg_attr(not(feature = "std"), no_std, no_main)]
+
+use crate::{errors::ShielderError, types::Scalar, contract::OpPub, merkle::{self, compute_hash}};
+
+trait Hashable {
+    fn hash(&self) -> Scalar;
+}
+
+#[derive(Clone, Copy, scale::Encode, scale::Decode)]
+#[cfg_attr(
+    feature = "std", 
+    derive(scale_info::TypeInfo)
+)]
+
+// empty private operation
+struct OpPriv {
+}
+
+#[derive(Clone, Copy, scale::Encode, scale::Decode)]
+#[cfg_attr(
+    feature = "std", 
+    derive(scale_info::TypeInfo)
+)]
+
+struct Operation {
+    op_pub: OpPub,
+}
+
+impl Operation {
+    fn combine(op_pub: OpPub, _op_priv: OpPriv) -> Self {
+        Operation{op_pub}
+    }
+}
+
+#[derive(Clone, Copy, scale::Encode, scale::Decode)]
+#[cfg_attr(
+    feature = "std", 
+    derive(scale_info::TypeInfo)
+)]
+struct Note {
+    id: Scalar,
+    trapdoor: Scalar,
+    nullifier: Scalar,
+    account_hash: Scalar,
+}
+
+impl Hashable for Note {
+    fn hash(&self) -> Scalar {
+        merkle::compute_hash(
+            self.id,
+            merkle::compute_hash(
+                self.trapdoor,
+                merkle::compute_hash(
+                    self.nullifier, 
+                    self.account_hash,
+                )
+            )
+        )
+    }
+}
+
+#[derive(Clone, Copy, scale::Encode, scale::Decode)]
+#[cfg_attr(
+    feature = "std", 
+    derive(scale_info::TypeInfo)
+)]
+struct Account {
+    balance_aleph: Scalar,
+    balance_usdt: Scalar,
+}
+
+impl Hashable for Account {
+    fn hash(&self) -> Scalar {
+        merkle::compute_hash(
+            self.balance_aleph,
+
+            self.balance_usdt
+        )
+    }
+}
+
+const USDT_TOKEN: [u8; 32] = [0x2_u8; 32];
+
+impl Account {
+    // TODO: increase and decrease balances
+    fn update(&self, operation: Operation) -> Self {
+        match operation.op_pub {
+            OpPub::Deposit(amount, token, _) => {
+                let balance_usdt = self.balance_usdt;
+                if token.as_ref() == USDT_TOKEN {
+                    // decrease scalar by amount
```

Comment:
You can just increase here, right? Or is there a reason this is still TODO?

Reply:
yes, wrote the whole module for mocked_zk